src/main/interface/msp.c

```diff
@@ -513,7 +513,7 @@ static bool mspCommonProcessOutCommand(uint8_t cmdMSP, sbuf_t *dst, mspPostProce
     case MSP_VOLTAGE_METERS: {
         // write out id and voltage meter values, once for each meter we support
         uint8_t count = supportedVoltageMeterCount;
-#ifndef USE_OSD_SLAVE
+#if !defined(USE_OSD_SLAVE) && defined(USE_ESC_SENSOR)
         count = supportedVoltageMeterCount - (VOLTAGE_METER_ID_ESC_COUNT - getMotorCount());
```

Comment:
This is probably better reformed to `count -= VOLTAGE_METER_ID_ESC_COUNT - getMotorCount()`.